CI warning is causing build to fail

The following shows up in some recent jobs: https://github.com/pandas-dev/pandas/actions/runs/3641110250/jobs/6146658226
```
2022-12-07T17:13:17.4282435Z pandas/_libs/tslibs/parsing.c: In function ‘__pyx_f_6pandas_5_libs_6tslibs_7parsing_dateutil_parse.constprop’:
2022-12-07T17:13:17.4286489Z pandas/_libs/tslibs/parsing.c:10634:22: error: array subscript ‘PyMethodObject[0]’ is partly outside array bounds of ‘PyObject[1]’ {aka ‘struct _object[1]’} [-Werror=array-bounds]
2022-12-07T17:13:17.4287232Z 10634 |           __pyx_t_10 = PyMethod_GET_SELF(__pyx_t_2);
2022-12-07T17:13:17.4288524Z In file included from /home/runner/micromamba/envs/test/include/python3.8/pytime.h:6,
2022-12-07T17:13:17.4289063Z                  from /home/runner/micromamba/envs/test/include/python3.8/Python.h:85,
2022-12-07T17:13:17.4289520Z                  from pandas/_libs/tslibs/parsing.c:46:
2022-12-07T17:13:17.4290230Z /home/runner/micromamba/envs/test/include/python3.8/object.h:560:22: note: while referencing ‘_Py_NoneStruct’
2022-12-07T17:13:17.4290904Z   560 | PyAPI_DATA(PyObject) _Py_NoneStruct; /* Don't use this directly */
2022-12-07T17:13:17.4291327Z       |                      ^~~~~~~~~~~~~~
2022-12-07T17:13:17.4299214Z pandas/_libs/tslibs/parsing.c:10636:23: error: array subscript ‘PyMethodObject[0]’ is partly outside array bounds of ‘PyObject[1]’ {aka ‘struct _object[1]’} [-Werror=array-bounds]
2022-12-07T17:13:17.4299878Z 10636 |             PyObject* function = PyMethod_GET_FUNCTION(__pyx_t_2);
2022-12-07T17:13:17.4300282Z       |                       ^~~~~~~~
2022-12-07T17:13:17.4300776Z In file included from /home/runner/micromamba/envs/test/include/python3.8/pytime.h:6,
2022-12-07T17:13:17.4301291Z                  from /home/runner/micromamba/envs/test/include/python3.8/Python.h:85,
2022-12-07T17:13:17.4301779Z                  from pandas/_libs/tslibs/parsing.c:46:
2022-12-07T17:13:17.4302447Z /home/runner/micromamba/envs/test/include/python3.8/object.h:560:22: note: while referencing ‘_Py_NoneStruct’
2022-12-07T17:13:17.4303287Z   560 | PyAPI_DATA(PyObject) _Py_NoneStruct; /* Don't use this directly */
2022-12-07T17:13:17.4303722Z       |                      ^~~~~~~~~~~~~~
2022-12-07T17:13:18.5474602Z cc1: all warnings being treated as errors
```

Example of green job: https://github.com/pandas-dev/pandas/actions/runs/3633476581/jobs/6130537543
Example of red job: https://github.com/pandas-dev/pandas/actions/runs/3641110250/jobs/6146658226

Can't tell what's caused this

Here's the difference between the environments:
```
                                                  Name Version_x Version_y
0    ──────────────────────────────────────────────...      None      None
59                                           db-dtypes     1.0.4     1.0.5
109                               google-cloud-storage     2.6.0     2.7.0
114                                             grpcio    1.49.1    1.51.1
115                                      grpcio-status    1.49.1    1.51.1
126                                importlib_resources    5.10.0    5.10.1
183                                            libgrpc    1.49.1    1.51.1
284                                     prompt-toolkit    3.0.33    3.0.36
```

--- a/.github/workflows/32-bit-linux.yml
+++ b/.github/workflows/32-bit-linux.yml
@@ -17,7 +17,7 @@ permissions:
 
 jobs:
   pytest:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/assign.yml
+++ b/.github/workflows/assign.yml
@@ -11,7 +11,7 @@ jobs:
     permissions:
       issues: write
       pull-requests: write
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     steps:
     - if: github.event.comment.body == 'take'
       run: |

--- a/.github/workflows/asv-bot.yml
+++ b/.github/workflows/asv-bot.yml
@@ -21,7 +21,7 @@ jobs:
     name: "Run benchmarks"
     # TODO: Support more benchmarking options later, against different branches, against self, etc
     if: startsWith(github.event.comment.body, '@github-actions benchmark')
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     defaults:
       run:
         shell: bash -el {0}

--- a/.github/workflows/autoupdate-pre-commit-config.yml
+++ b/.github/workflows/autoupdate-pre-commit-config.yml
@@ -15,7 +15,7 @@ jobs:
       pull-requests: write  # for technote-space/create-pr-action to create a PR
     if: github.repository_owner == 'pandas-dev'
     name: Autoupdate pre-commit config
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     steps:
       - name: Set up Python
         uses: actions/setup-python@v4

--- a/.github/workflows/code-checks.yml
+++ b/.github/workflows/code-checks.yml
@@ -20,7 +20,7 @@ permissions:
 jobs:
   pre_commit:
     name: pre-commit
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     concurrency:
       # https://github.community/t/concurrecy-not-work-for-push/183068/7
       group: ${{ github.event_name == 'push' && github.run_number || github.ref }}-pre-commit
@@ -41,7 +41,7 @@ jobs:
 
   docstring_typing_pylint:
     name: Docstring validation, typing, and pylint
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     defaults:
       run:
         shell: bash -el {0}
@@ -104,7 +104,7 @@ jobs:
 
   asv-benchmarks:
     name: ASV Benchmarks
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     defaults:
       run:
         shell: bash -el {0}
@@ -135,7 +135,7 @@ jobs:
 
   build_docker_dev_environment:
     name: Build Docker Dev Environment
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     defaults:
       run:
         shell: bash -el {0}
@@ -162,7 +162,7 @@ jobs:
 
   requirements-dev-text-installable:
     name: Test install requirements-dev.txt
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
 
     concurrency:
       # https://github.community/t/concurrecy-not-work-for-push/183068/7

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -10,7 +10,7 @@ concurrency:
 
 jobs:
   analyze:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     permissions:
       actions: read
       contents: read

--- a/.github/workflows/docbuild-and-upload.yml
+++ b/.github/workflows/docbuild-and-upload.yml
@@ -22,7 +22,7 @@ permissions:
 jobs:
   web_and_docs:
     name: Doc Build and Upload
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
 
     concurrency:
       # https://github.community/t/concurrecy-not-work-for-push/183068/7

--- a/.github/workflows/package-checks.yml
+++ b/.github/workflows/package-checks.yml
@@ -17,7 +17,7 @@ permissions:
 jobs:
   pip:
     if: ${{ github.event.label.name == 'Build' || contains(github.event.pull_request.labels.*.name, 'Build') || github.event_name == 'push'}}
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     strategy:
       matrix:
         extra: ["test", "performance", "timezone", "computation", "fss", "aws", "gcp", "excel", "parquet", "feather", "hdf5", "spss", "postgresql", "mysql", "sql-other", "html", "xml", "plot", "output_formatting", "clipboard", "compression", "all"]

--- a/.github/workflows/python-dev.yml
+++ b/.github/workflows/python-dev.yml
@@ -49,7 +49,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macOS-latest, windows-latest]
+        os: [ubuntu-20.04, macOS-latest, windows-latest]
 
     name: actions-311-dev
     timeout-minutes: 120

--- a/.github/workflows/scorecards.yml
+++ b/.github/workflows/scorecards.yml
@@ -13,7 +13,7 @@ permissions: read-all
 jobs:
   analysis:
     name: Scorecards analysis
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     permissions:
       # Needed to upload the results to code-scanning dashboard.
       security-events: write

--- a/.github/workflows/sdist.yml
+++ b/.github/workflows/sdist.yml
@@ -19,7 +19,7 @@ permissions:
 jobs:
   build:
     if: ${{ github.event.label.name == 'Build' || contains(github.event.pull_request.labels.*.name, 'Build') || github.event_name == 'push'}}
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     timeout-minutes: 60
     defaults:
       run:

--- a/.github/workflows/stale-pr.yml
+++ b/.github/workflows/stale-pr.yml
@@ -11,7 +11,7 @@ jobs:
   stale:
     permissions:
       pull-requests: write
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     steps:
     - uses: actions/stale@v4
       with:

--- a/.github/workflows/ubuntu.yml
+++ b/.github/workflows/ubuntu.yml
@@ -20,7 +20,7 @@ permissions:
 
 jobs:
   pytest:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     defaults:
       run:
         shell: bash -el {0}

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -127,7 +127,7 @@ jobs:
       (github.event_name == 'pull_request' &&
       contains(github.event.pull_request.labels.*.name, 'Build')) ||
       (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v') && ( ! endsWith(github.ref, 'dev0')))
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     env:
       IS_PUSH: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v') }}
       IS_SCHEDULE_DISPATCH: ${{ github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' }}

